src/modules/esl-event-listener/core/targets/swipe.target.ts

```diff
@@ -0,0 +1,180 @@
+import {SyntheticEventTarget} from '../../../esl-utils/dom/events/target';
+import {ESLMixinElement} from '../../../esl-mixin-element/ui/esl-mixin-element';
+import {overrideEvent} from '../../../esl-utils/dom/events/misc';
+
+type SwipeDirection = 'left' | 'right' | 'up' | 'down';
+type SwipeEventName = 'swipe' | 'swipe:left' | 'swipe:right' | 'swipe:up' | 'swipe:down';
+
+interface SwipeEvent {
```

Comment:
We can make an exact Event class to provide both typings and creation incapsulation.

See `resize.adapter.event.ts`, `resize.adapter.ts` , and `decorated.target.ts` implementations, as they will have a lot of similarities in some implementation moments.